zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateMessageEventSubprocessTest.java

```diff
@@ -0,0 +1,596 @@
+        .hasElementId("start2")
+        .describedAs("Expect that the correlation key is not re-evaluated")
+        .hasCorrelationKey(helper.getCorrelationValue());
+
+    assertThat(
+            RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withMessageName("msg")
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that the process definition is updated")
+        .hasBpmnProcessId(targetProcessId)
+        .describedAs("Expect that the correlation key is not re-evaluated")
+        .hasCorrelationKey(helper.getCorrelationValue());
+
+    ENGINE.message().withName("msg").withCorrelationKey(helper.getCorrelationValue()).publish();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementId("eventSubprocessUserTask")
+                .getFirst())
+        .describedAs(
+            "Expect that the element inside the event subprocess in the target process is activated")
+        .isNotNull();
+  }
+
+  @Test
+  public void shouldUnsubscribeFromMessageEventSubprocess() {
+    // given
+    final String processId = helper.getBpmnProcessId();
+    final String targetProcessId = helper.getBpmnProcessId() + "2";
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .eventSubProcess(
+                        "sub",
+                        s ->
+                            s.startEvent("start1")
+                                .message(m -> m.name("msg").zeebeCorrelationKeyExpression("key1"))
+                                .serviceTask("A", t -> t.zeebeJobType("task"))
+                                .endEvent())
+                    .startEvent("start")
+                    .userTask("userTask1")
+                    .endEvent("end")
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent("start")
+                    .userTask("userTask2")
+                    .endEvent("end")
+                    .done())
+            .deploy();
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withVariable("key1", helper.getCorrelationValue())
+            .create();
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("userTask1", "userTask2")
+        .migrate();
```

Comment:
❌Please await the activation of the user task to ensure the test is deterministic